components/omega/doc/design/VerticalMixingCoeff.md

```diff
@@ -0,0 +1,182 @@
+
+```c++
+parallelFor(
+    {NCellsAll, NVertLevels}, KOKKOS_LAMBDA(int ICell, int K) {
+
+        VertViscosity(ICell, K) = 0._Real;
+        VertDiffusivity(ICell, K) = 0._Real;
+
+        // If shear mixing true, add shear contribution to viscosity and diffusivity 
+        if (EnableShearMix) {
+            const Real InvAreaCell = 1._Real / AreaCell(ICell);
+            // Calculate the square of the shear
+            for (int J = 0; J < NEdgesOnCell(ICell); ++J) {
+                const I4 JEdge = EdgesOnCell(ICell, J);
+                const Real Factor = 0.5_Real * DcEdge(JEdge) * DvEdge(JEdge) * InvAreaCell;
+                const Real DelU2 = pow(NormalVelocity(JEdge,K-1) - NormalVelocity(JEdge,K), 2._Real) + pow(TangentialVelocity(JEdge,K-1) - TangentialVelocity(JEdge,K), 2._Real);
+                ShearSquared(ICell, K) = ShearSquared(ICell, K) + Factor * DelU2;
+            }
+            ShearSquared(ICell, K) = ShearSquared(ICell, K) / pow(zMid(ICell, K-1) - zMid(ICell, K), 2._Real);
+
+            // Calculate Richardson number
+            RichardsonNum(ICell, K) = BruntVaisalaFreq(ICell, K) / max(ShearSquared(ICell, K), 1.0e-12_Real);
+
+            // Add in shear contribution to vertical mixing
+            VertViscosity(ICell, K) = VertViscosity(ICell, K) + ShearNuZero / pow(1._Real + ShearAlpha * RichardsonNum(ICell, K), ShearExponent);
+            VertDiffusivity(ICell, K) = VertDiffusivity(ICell, K) + VertViscosity(ICell, K) / (1 + ShearAlpha * RichardsonNum(ICell, K));
+        }
+
+        // If conv mixing true, add conv contribution to viscosity and diffusivity 
+        if (EnableConvectiveMix) {
+            if (BruntVaisalaFreq(ICell, K) < ConvectiveTriggerBVF) {
+                VertViscosity(ICell, K) = VertViscosity(ICell, K) + ConvectiveDiffusivity;
+                VertDiffusivity(ICell, K) = VertDiffusivity(ICell, K) + ConvectiveDiffusivity;
+            }
+        }
+
+        // Add background contribution to viscosity and diffusivity 
+        VertViscosity(ICell, K) = VertViscosity(ICell, K) + BackgroundViscosity;
+        VertDiffusivity(ICell, K) = VertDiffusivity(ICell, K) + BackgroundDiffusivity;
+    });
+```
+
+From here, the vertical viscosity and vertical diffusivity will enter into the [tridiagonal solver](./TridiagonalSolver.md) to compute the vertical flux of momentum and tracers.
+
+## 5 Verification and Testing
+
+Unit tests can be initialized with linear-with-depth initial conditions (velocity and density) and use a linear equation of state. Expected values of the Richardson number, vertical viscosity, and vertical diffusion coefficients can be computed and compared to. Tests for both the shear, convective, and combined mixing contributions should be made to test requirement (2.1).
```

Comment:
Let me know if you think polaris tests would be helpful. Seems like unit tests could be sufficient until we increase complexity but perhaps a decomp test could be helpful in polaris.

Reply:
I agree. Testing each process sounds like a good unit test. Once you've verified the combined mixing with unit tests, we can add the analogous test to Polaris for Polaris code coverage.

Reply:
Great, sounds good! Thanks